[BUG] invalid memory address or nil pointer dereference

**panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x81aafa]**

mubeng works correctly when we use in sqlmap nuclei and etc....
BUT some tools like Acunetix, Appscan when use proxy server, mubeng crushes

I guess maybe some packet corrupt but mubeng not handle it. please follow screenshot

**Screenshots**
![image](https://github.com/kitabisa/mubeng/assets/156658310/1b215463-8c0b-4dc9-93f3-066882b2889b)


- OS: linux
- OS version: Linux 5.15.0-91-generic 
- mubeng Version: v0.14.1

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x81aafa]

goroutine 38 [running]:
ktbs.dev/mubeng/internal/server.(*Proxy).onRequest.func1()
        /app/internal/server/handler.go:45 +0x5a
created by ktbs.dev/mubeng/internal/server.(*Proxy).onRequest
        /app/internal/server/handler.go:44 +0x37d

--- a/internal/server/handler.go
+++ b/internal/server/handler.go
@@ -41,13 +41,13 @@ func (p *Proxy) onRequest(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Reque
 	rotate = helper.EvalFunc(rotate)
 	resChan := make(chan interface{})
 
-	go func() {
-		if (req.URL.Scheme != "http") && (req.URL.Scheme != "https") {
-			resChan <- fmt.Errorf("Unsupported protocol scheme: %s", req.URL.Scheme)
+	go func(r *http.Request) {
+		if (r.URL.Scheme != "http") && (r.URL.Scheme != "https") {
+			resChan <- fmt.Errorf("Unsupported protocol scheme: %s", r.URL.Scheme)
 			return
 		}
 
-		log.Debugf("%s %s %s", req.RemoteAddr, req.Method, req.URL)
+		log.Debugf("%s %s %s", r.RemoteAddr, r.Method, r.URL)
 
 		tr, err := mubeng.Transport(rotate)
 		if err != nil {
@@ -82,7 +82,7 @@ func (p *Proxy) onRequest(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Reque
 		resp.Body = io.NopCloser(bytes.NewBuffer(buf))
 
 		resChan <- resp
-	}()
+	}(req)
 
 	var resp *http.Response
 

